backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -3050,4 +3051,33 @@ public ExperimentRun cloneExperimentRun(
     desExperimentRunBuilder.clearOwner().setOwner(authService.getVertaIdFromUserInfo(userInfo));
     return insertExperimentRun(projectDAO, desExperimentRunBuilder.build(), userInfo);
   }
+
+  @Override
+  public void logEnvironment(String experimentRunId, EnvironmentBlob environmentBlob)
+      throws InvalidProtocolBufferException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      Transaction transaction = session.beginTransaction();
+      ExperimentRunEntity experimentRunEntity =
+          session.get(ExperimentRunEntity.class, experimentRunId);
+      if (experimentRunEntity == null) {
+        LOGGER.info(ModelDBMessages.EXP_RUN_NOT_FOUND_ERROR_MSG);
+        Status status =
+            Status.newBuilder()
+                .setCode(Code.NOT_FOUND_VALUE)
+                .setMessage(ModelDBMessages.EXP_RUN_NOT_FOUND_ERROR_MSG)
+                .build();
+        throw StatusProto.toStatusRuntimeException(status);
+      }
+      experimentRunEntity.setEnvironment(ModelDBUtils.getStringFromProtoObject(environmentBlob));
+      session.update(experimentRunEntity);
+      transaction.commit();
+      LOGGER.debug("EnvironmentBlob logged successfully");
+    } catch (Exception ex) {
```

Comment:
Where is the permission check being done? I only see direct DB operations.

Reply:
Ohh Sorry, i forget to add it.

Reply:
Now added check on the service https://github.com/VertaAI/modeldb/pull/1786/files#diff-73f4a3ee85b589c5a06a8200ac022bf55ffc3f201c9b4abe78bd9734b946da44R2806